### PR TITLE
KOGITO-4162: Doc - Create Management/Task Console quickstart sample project with auth

### PR DESCRIPTION
**Pull request by pefernan**
Quickstart with Auth!

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `master` branch!**

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: kogito-quickstarts/02-kogito-auth-example/README.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+> For more information about authorization on Quarkus applications with OpenID Connect, please refer to [Quarkus Documentation](https://quarkus.io/guides/security-openid-connect-web-authentication)
+
+## Running the Quickstart
+
+### Prerequisites
+
+* Java 11+ installed
+* Environment variable JAVA_HOME set accordingly
+* Maven 3.5.2+ installed
+* Docker and Docker Compose to run the required example infrastructure.
```
Comment (nmirasch) [inline kogito-quickstarts/02-kogito-auth-example/README.md:77]: @pefernan  here we need to add this as prerequisite regarding the commands you are using to generate the bearer token:
" - [jq](https://stedolan.github.io/jq) tool installed. You can download it from [here](https://stedolan.github.io/jq/download)"

File: kogito-quickstarts/01-kogito-basic-example/README.md [hunk 2]
```diff
@@ -70,19 +70,29 @@ cd <path_to_kogito_quickstart>/
 mvn clean package quarkus:dev
 ```
 
-Once the service is up and running you can POST to `http://localhost:8080/hiring` to start creating instances of the 
-**Hiring**.
+### Starting a Hiring Process
 
-For example on a Terminal by executing:
+Once the service is up and running you can making use of the **Hiring** application by a sending request to `http://localhost:8080/hiring`  with following content:
```
Comment (sterobin) [inline kogito-quickstarts/01-kogito-basic-example/README.md]: "can make use of"

File: kogito-quickstarts/01-kogito-basic-example/docker-compose/README.md [hunk 3]
```diff
@@ -43,7 +43,7 @@ In order to use it, please ensure you have Docker Compose installed on your mach
 
     docker-compose stop
     
-  It is also recommended to remove any of stopped containers by running:
+  It is also recommended removing any of stopped containers by running:
```
Comment (sterobin) [inline kogito-quickstarts/01-kogito-basic-example/docker-compose/README.md]: Actually the previous version was correct, "recommended to remove"

File: kogito-quickstarts/02-kogito-auth-example/README.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
+# 02.- Kogito Example with Authentication
+
+## Description
+
+In the [Basic Example](../01-kogito-basic-example) we saw a basic setup for the **Hiring** process using. In this Quickstart we are going to enhance it adding security to our setup using [Keycloak](https://www.keycloak.org/).
```
Comment (sterobin) [inline kogito-quickstarts/02-kogito-auth-example/README.md]: "process using ___" ? Missing something?

File: kogito-quickstarts/02-kogito-auth-example/README.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+In this Quickstart we'll be using [Keycloak](https://www.keycloak.org/) as *Authentication Server*. It will be started as a part of the project *Infrastructure Services*, you can check the configuration on the project [docker-compose.yml](./docker-compose/docker-compose.yml) in [docker-compose](./docker-compose) folder.
+
+It will install the *Kogito Realm* that comes with a predefined set of users:
+| Login         | Password   | Roles               |
+| ------------- | ---------- | ------------------- |
+|    admin      |   admin    | *admin*, *HR*, *IT* |
+|    alice      |   alice    | *HR*                |
+|    jdoe       |   jdoe     | *IT*                |
+
+Once started, you should be able to access your *Keycloak Server* at [localhost:8480/auth](http://localhost:8480/auth) with *admin* user.
```
Comment (sterobin) [inline kogito-quickstarts/02-kogito-auth-example/README.md]: "Once the server is started, ..."

File: kogito-quickstarts/02-kogito-auth-example/README.md [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+quarkus.oidc.client-id=kogito-app
+quarkus.oidc.credentials.secret=secret
+
+quarkus.http.auth.permission.authenticated.paths=/*
+quarkus.http.auth.permission.authenticated.policy=authenticated
+```
+
+> **_NOTE:_**  For more information about securing Quarkus applications with OpenID Connect, please refer to [Quarkus Documentation](https://quarkus.io/guides/security-openid-connect)
+
+It is also possible enabling authentication *Kogito Management Console* & *Kogito Task Console* using OpenID Connect, to do that you must start the consoles using the *keycloak* quarkus profile as a parameter (`-Dquarkus.profile=keycloak`). You can check the configuration on the project [docker-compose.yml](./docker-compose/docker-compose.yml) in [docker-compose](./docker-compose) folder.
```
Comment (sterobin) [inline kogito-quickstarts/02-kogito-auth-example/README.md]: "possible to enable authentication ..."

File: kogito-quickstarts/02-kogito-auth-example/README.md [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+* Environment variable JAVA_HOME set accordingly
+* Maven 3.5.2+ installed
+* [jq](https://stedolan.github.io/jq) tool installed. You can download it from [here](https://stedolan.github.io/jq/download)
+* Docker and Docker Compose to run the required example infrastructure.
+
+### Starting the Kogito and Infrastructure Services
+
+This quickstart provides a docker compose template that starts all the required services. This setup ensures that all services are connected with a default configuration.
+
+You should start all the services before you execute any of the **Hiring** example, to do that please execute:
```
Comment (sterobin) [inline kogito-quickstarts/02-kogito-auth-example/README.md]: "Before you execute ... example, start all the services by following these steps:"

File: kogito-quickstarts/02-kogito-auth-example/README.md [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+- Infinispan: 11222
+- Kafka: 9092
+- Keycloak: 8480
+- Data Index: 8180
+- Management Console: 8280
+- Task Console: 8380
+
+> **_NOTE:_**  This step requires the project to be compiled, please consider runing a ```mvn clean compile``` command on the project root before running the ```startServices.sh``` script for the first time or any time you modify the project.
+
+Once started you can simply stop all services by executing the ```docker-compose stop```.
```
Comment (sterobin) [inline kogito-quickstarts/02-kogito-auth-example/README.md]: "To stop all services, execute ..."

File: kogito-quickstarts/02-kogito-auth-example/README.md [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+cd <path_to_kogito_quickstart>/
+```
+3.- Start the example with the command
+```bash
+mvn clean package quarkus:dev
+```
+
+### Starting a Hiring Process
+
+Once the service is up and running you can making use of the **Hiring** application by a sending request to `http://localhost:8080/hiring`  with following content:
```
Comment (sterobin) [inline kogito-quickstarts/02-kogito-auth-example/README.md]: "can make use of"

File: kogito-quickstarts/02-kogito-auth-example/README.md [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+
+### Kogito Management Console
+
+To access the Kogito Management Console just open your browser and navigate to ``http://localhost:8280``. You'll be redirected to the *Keycloak* login page
+
+<p align="center">
+    <img width=75%  src="docs/images/keycloak-login.png">
+</p>
+
+Once there, login using any of the users specified in the [Using Keycloak as Authentication Server](#using-keycloak-as-authentication-server)
```
Comment (sterobin) [inline kogito-quickstarts/02-kogito-auth-example/README.md]: "log in" two words here

File: kogito-quickstarts/02-kogito-auth-example/README.md [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+
+### Kogito Task Console
+
+To access the Kogito Task Console just open your browser and navigate to ``http://localhost:8380``. You'll be redirected to the *Keycloak* login page
+
+<p align="center">
+    <img width=75%  src="docs/images/keycloak-login.png">
+</p>
+
+Once there, login with an *HR* user (for example *alice*) and you should be redirected to the user **Task Inbox**:
```
Comment (sterobin) [inline kogito-quickstarts/02-kogito-auth-example/README.md]: "log in"

File: kogito-quickstarts/02-kogito-auth-example/docker-compose/README.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+- Kogito Task Console
+
+This setup ensures that all services are connected using the default configuration as well as provisioning the Travel Agency dashboard to Grafana.  
+
+In order to use it, please ensure you have Docker Compose installed on your machine, otherwise follow the instructions available
+ in [here](https://docs.docker.com/compose/install/).
+ 
+### Starting required services
+
+  You should start all the services before you execute any of the Travel Agency applications, to do that please execute:
```
Comment (sterobin) [inline kogito-quickstarts/02-kogito-auth-example/docker-compose/README.md]: "Before you execute ..., start all services by following these steps:"

File: kogito-quickstarts/02-kogito-auth-example/docker-compose/README.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+  - Management Console: 8280
+  - Task Console: 8380
+
+### Stopping and removing volume data
+  
+  To stop all services, simply run:
+
+    docker-compose stop
+    
+  It is also recommended removing any of stopped containers by running:
```
Comment (sterobin) [inline kogito-quickstarts/02-kogito-auth-example/docker-compose/README.md]: "recommended to remove"

**Review by sterobin — COMMENTED** [review]
@pefernan Comments inline after a brief look-through, all minor things.

**Review by cristianonicolai — APPROVED** [review]

**Review by nmirasch — APPROVED** [review]

**Review by AjayJagan — APPROVED** [review]
LGTM :)

**Review by Sara4994 — APPROVED** [review]

**Comment by pefernan** [comment]
@sterobin are we ok to merge??

**Pull request merged**